packages/create-amplify/src/logger.ts

```diff
@@ -1,63 +1,150 @@
 import yargs from 'yargs';
+import * as os from 'os';
 
 /**
  * A logger that logs messages to the console.
  */
 export class Logger {
+  // Properties for ellipsis animation
+  private timer: ReturnType<typeof setTimeout>;
+  private refreshRate: number;
+  private timerSet: boolean;
+
   /**
-   * Creates a new Logger instance.
-   * @param console The console to log to.
+   * Creates a new Logger instance. Injecting stdout for testing.
+   * @param stdout The stream to write logs to.
    * @param minimumLogLevel The minimum log level to log.
    */
   constructor(
-    private readonly console: Console = global.console,
+    private readonly stdout = process.stdout,
     private readonly minimumLogLevel: LogLevel = LogLevel.INFO
-  ) {}
+  ) {
+    this.refreshRate = 500; // every 0.5 seconds
+  }
 
   /**
    * Logs a message to the console.
    */
-  async log(message: string, level: LogLevel = LogLevel.INFO) {
+  log(message: string, level: LogLevel = LogLevel.INFO) {
     const toLogMessage = level <= this.minimumLogLevel;
 
     if (!toLogMessage) {
       return;
     }
 
-    const logMessage = `[${
-      LogLevel[level]
-    }] ${new Date().toISOString()}: ${message}`;
-    this.console.log(logMessage);
+    const logMessage =
+      this.minimumLogLevel === LogLevel.DEBUG
+        ? `[${LogLevel[level]}] ${new Date().toISOString()}: ${message}`
+        : message;
+    this.stdout.write(logMessage + os.EOL);
+  }
+
+  /**
+   * Logs a message with animated ellipsis
+   */
+  async withEllipsis(message: string, callback: () => Promise<void>) {
```

Comment:
Do you think we'll ever need different log levels here?
(I think that non-info messages with ellipsis would be very rare, so perhaps not, but please give it a thought)